packages/pds/src/api/com/atproto/server/createAccount.ts

```diff
@@ -17,60 +17,10 @@ export default function (server: Server, ctx: AppContext) {
       points: 100,
     },
     handler: async ({ input, req }) => {
-      const { email, password, inviteCode } = input.body
-      if (!email) {
-        throw new InvalidRequestError('Missing input: "email"')
-      } else if (!password) {
-        throw new InvalidRequestError('Missing input: "password"')
-      } else if (input.body.plcOp) {
-        throw new InvalidRequestError('Unsupported input: "plcOp"')
-      }
-
-      if (ctx.cfg.invites.required && !inviteCode) {
-        throw new InvalidRequestError(
-          'No invite code provided',
-          'InvalidInviteCode',
-        )
-      }
-
-      if (!disposable.validate(email)) {
-        throw new InvalidRequestError(
-          'This email address is not supported, please use a different email.',
-        )
-      }
-
-      // normalize & ensure valid handle
-      const handle = await normalizeAndValidateHandle({
-        ctx,
-        handle: input.body.handle,
-        did: input.body.did,
-      })
-
-      // check that the invite code still has uses
-      if (ctx.cfg.invites.required && inviteCode) {
-        await ctx.accountManager.ensureInviteIsAvailable(inviteCode)
-      }
-
-      // check that the handle and email are available
-      const [handleAccnt, emailAcct] = await Promise.all([
-        ctx.accountManager.getAccount(handle),
-        ctx.accountManager.getAccountByEmail(email),
-      ])
-      if (handleAccnt) {
-        throw new InvalidRequestError(`Handle already taken: ${handle}`)
-      } else if (emailAcct) {
-        throw new InvalidRequestError(`Email already taken: ${email}`)
-      }
-
-      // determine the did & any plc ops we need to send
-      // if the provided did document is poorly setup, we throw
-      const signingKey = await Secp256k1Keypair.create({ exportable: true })
-      const { did, plcOp } = await getDidAndPlcOp(
-        ctx,
-        handle,
-        input.body,
-        signingKey,
-      )
+      const { did, handle, email, password, inviteCode, signingKey, plcOp } =
+        ctx.entrywayAgent
+          ? await validateInputsForEntrywayPds(ctx, input.body)
+          : await validateInputsForLocalPds(ctx, input.body)
```

Comment:
This ternary is backwards! Or rather, it logically does the right thing, but the names of the two functions are backwards. Unless I'm mistaken, `validateInputsForEntrywayPds` contains the Local PDS validation logic, and vice versa. The functions should perhaps have their names swapped?

Reply:
You're right, it's confusing! Will take a look at fixing the names, adding better comments 👍 There are three different setups kicking around, and this package handles two of them:

 - Being an entryway (not covered here)
 - Being a pds behind an entryway, e.g. the mushroom servers (validateInputsForEntrywayPds())
 - Being a pds that doesn't use an entryway, e.g. a self-hoster (validateInputsForLocalPds())

Reply:
Ahh that makes sense, thanks